### DISCUSSION TITLE
Using jQuery with Livewire 3

### DISCUSSION THREAD

**Discussion by wcoy7** [livewire/livewire#7896] [1. Help]
With jQuery the $ is used by it, how can I tell Alpine to use something besides $wire?

**Comment by danie-ramdhani** [comment]
you can still use `$wire` object because it's just object, not function. and jquery selector is look like this:

```js
$(...) //function
$('.my-class')
```
if you want to use alternative syntax for `$wire`, use `@this`:

```blade
<span x-text="@this.myProp"></span>
```

> (is Alpine js not enough for you or you using some lib that require jquery?)

**Comment by wcoy7** [comment]
jQuery doesn't really work that way.   What I wanted to do is listen to a select2 event like [select2:select](https://select2.org/programmatic-control/events).  Maybe that's what I am doing wrong, I should rewrite the event listener in AlpineJS?

**Comment by wcoy7** [comment]
What I want to do is listen to the event, grab whatever data was selected, dispatch that to the component, and re-initialize the select2 (clear the value and fire it's change event). Normally I would just write a javascript function and attach that to a listener.   I think I worked around it by doing Livewire.dispatch as $wire is Uncaught ReferenceError: $wire is not defined

**Reply by Kotzilla** [reply]
for select2 i guess you can add  push script at the end of component 

```
@push('scripts')
  <script>
    $(document).ready(function() {
      $('#state_id').on('select2:select', function (e) {
        @this.set('state_id', e.params.data.id);
      });
    });
  </script>
@endpush
```

**Comment by wcoy7** [comment]
When I do Alpine it shows the version number, $wire is UncaughtReferenceError.   In this project I have not setup jQuery yet.
